dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/query/SearchForTypes.java

```diff
@@ -62,6 +62,18 @@ public SearchForTypes toggleType(SearchObjectType type) {
         return new SearchForTypes(newTypes);
     }
     
+    public SearchForTypes addType(SearchObjectType type) {
+        Set<SearchObjectType> newTypes = new HashSet<>(types);
+        newTypes.add(type);
+        return new SearchForTypes(newTypes);
+    }
+    
+    public SearchForTypes removeType(SearchObjectType type) {
+        Set<SearchObjectType> newTypes = new HashSet<>(types);
+        newTypes.remove(type);
+        return new SearchForTypes(newTypes);
+    }
```

Comment:
Can we have java docs for these? Or can you refactor names to indicate that new instance is being returned?

Reply:
At the end I didn't used this functions - so I will remove them